src/Native/Unix/configure.cmake

```diff
@@ -316,6 +316,10 @@ check_function_exists(
     fcopyfile
     HAVE_FCOPYFILE)
 
+check_function_exists(
```

Comment:
I rather meant checking for the sys/poll.h header presence to make sure we don't try to include it if it doesn't exist. Like this:
```cmake
check_include_files(
     "sys/poll.h"
     HAVE_SYS_POLL_H)
```

I am not sure if the poll function detection as you have it would work, since I think that cmake would not add the "sys/poll.h" include into its little testing source that it generates.